SETUP.md

```diff
@@ -62,53 +62,42 @@ You can do Code.org development using OSX, Ubuntu, or Windows (running Ubuntu in
-    - B. `rbenv`: ([instructions](https://github.com/sstephenson/rbenv#installation))
-        1. Install `rbenv` and `ruby-build`
-        1. `rbenv install 2.2.3`
-        1. `rbenv global 2.2.3`
-        1. `rbenv rehash`
-    - C. `rvm`. A few folks have had more luck with rvm vs rbenv on linux.
-        1. Install rvm from https://rvm.io/
-        1. `rvm install 2.2.3`
-        1. `rvm use 2.2.3 --default`
-1. Install Node, npm, and yarn
-  1. Option A - [nvm](https://github.com/creationix/nvm)
-    1. `curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.31.0/install.sh | bash`
-      - After completion, close your current terminal window and open a new one.
-    1. `nvm install` (this will install the version defined in `.nvmrc`)
-  1. Option B - nodesource repository
-    1. `curl -sL https://deb.nodesource.com/setup_0.12 | sudo -E bash -`
-    1. `sudo apt-get install -y nodejs`
-  1. Option C - Manual install
-    1. [Nodejs.org](https://nodejs.org/download/)
-  1. Finally, install yarn: `npm install -g yarn@0.23.2`
-1. When running `bundle install`, you may need to run
-
-   `bundle config build.nokogiri --use-system-libraries=true --with-xml2-include=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include/libxml2`
-
-   first so that nokogiri picks up the built in libxml2 library. See this [stackoverflow answer](http://stackoverflow.com/a/40038954/186771).
+1. `sudo apt-get install -y git mysql-server mysql-client libmysqlclient-dev libxslt1-dev libssl-dev zlib1g-dev imagemagick libmagickcore-dev libmagickwand-dev openjdk-9-jre-headless libcairo2-dev libjpeg8-dev libpango1.0-dev libgif-dev curl pdftk enscript libsqlite3-dev phantomjs build-essential redis-server`
+    * **Hit enter and select default options for any configuration popups, leaving mysql passwords blank**
+1. If it is not already installed, install Git with `sudo apt install git`
+1. Install npm with `sudo apt install npm`, then
+    * `sudo add-apt-repository ppa:chris-lea/node.js`
+    * `sudo apt-get update`
+    * `sudo apt-get install nodejs`
+1. Install Ruby 2.2.3 with rbenv
+     1. Install rbenv with `sudo apt install rbenv` and ruby-build with `sudo apt install ruby-build`
+     1. `rbenv install 2.2.3`
+     1. `rbenv global 2.2.3`
+     1. `rbenv rehash`
+1. Install Node and yarn
+    1. Node: `sudo apt install nodejs-legacy`
+    1. yarn: `sudo npm install -g yarn@0.23.2`
+1. Run `gem install bundler -v 1.14.6`. Don't worry about using version 1.10.6 from the overview
+1. Before running `bundle install`, make sure to run `sudo apt install ruby2.3-dev`
+1. Finally, configure your mysql to allow for a proper installation
+   1. If you properly left all the default options blank, type `sudo mysql` to start it up
+   1. Type `ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '';`
+   1. Then `FLUSH PRIVILEGES;`
+   1. And finally `exit;` to close out
```

Comment:
Also, thanks very much for including this step, this is a big issue in ubuntu-packaged mysql 5.7 with our application!

---

Note-to-self on what this query does and why it's needed (no doc changes needed, just adding details here for future reference):

This extra query on setup is needed to support a [change](https://anonscm.debian.org/cgit/pkg-mysql/mysql.git/tree/debian/NEWS) in Debian/Ubuntu packaging for `mysql >= 5.7.13-1`, which is the only mysql installation available by default for 16.04 and up (14.04 is the last release to package [`mysql-server-5.6`](https://packages.ubuntu.com/trusty/mysql-server-5.6); 16.04 and up use 5.7.x for [`mysql-server`](https://packages.ubuntu.com/xenial/mysql-server) packages):

> Password behaviour when the MySQL root password is empty has changed. Packaging now enables [socket authentication](https://dev.mysql.com/doc/refman/5.7/en/socket-pluggable-authentication.html) when the MySQL root password is empty. This means that a non-root user can't log in as the MySQL root user with an empty password.

Our application's default is to access the database as the MySQL `root` user with an empty password, so the extra `ALTER USER ... IDENTIFIED WITH` MySQL query reverts this new behavior back to the original native-password-based implementation ([`mysql_native_password`](https://dev.mysql.com/doc/refman/5.7/en/native-pluggable-authentication.html)).